apps/src/music/views/Generate.tsx

```diff
@@ -53,29 +59,63 @@ const Generate: React.FunctionComponent = () => {
     packId
   );
 
-  const generateSong = useCallback(() => {
+  const generateSongAi = useCallback(async () => {
     console.log('starting ask');
-    dispatch(setAiGenerateState('generating'));
-    askAi(
+    const result = await askAi(
       `Here is the context:
   ${contextGenerateMusicPsuedocodeFromDescription}
 
   And here is the request:
   ${useText}`
-    ).then(result => {
-      if (result.length > 1 && result[1].status === AiInteractionStatus.OK) {
-        console.log(result[1].chatMessageText);
-        const psuedocode = result[1].chatMessageText.replaceAll('```', '');
-        const resultBlockly = generateBlocklyJson(psuedocode);
-        dispatch(setCodeToLoad(resultBlockly));
-      } else {
-        console.error('Error getting AI response.');
-      }
-
-      dispatch(setAiGenerateState('done'));
-    });
+    );
+    if (result.length > 1 && result[1].status === AiInteractionStatus.OK) {
+      console.log(result[1].chatMessageText);
+      const psuedocode = result[1].chatMessageText.replaceAll('```', '');
+      const resultBlockly = generateBlocklyJson(psuedocode);
+      dispatch(setCodeToLoad(resultBlockly));
+    } else {
+      console.error('Error getting AI response.');
+    }
   }, [dispatch, contextGenerateMusicPsuedocodeFromDescription, useText]);
 
+  const delay = (time: number) => {
+    return new Promise(res => {
+      setTimeout(res, time);
+    });
+  };
+
+  const generateSongCache = useCallback(async () => {
+    const variant = getRandomInt(0, 2);
+    const cacheFilePath = `${baseAssetUrl}generate/music/${packId}-${adlibOption}-${joinedChoicesText}-${variant
+      .toString()
+      .padStart(2, '0')}.txt`;
+    console.log(cacheFilePath);
+
+    const startTime = Date.now();
+    try {
+      const response = await HttpClient.get(cacheFilePath);
+      const responseText = await response.text();
+      const elapsedTime = Date.now() - startTime;
+      const delayDuration = 2000; // 2 seconds.
+      const remainingDelayDuration = Math.max(delayDuration - elapsedTime, 0);
+      await delay(remainingDelayDuration);
```

Comment:
Could just do `await new Promise(r => setTimeout(r, remainingDelayDuration))` if you want an equivalent one-liner?